man/coredump.conf.xml

```diff
@@ -140,6 +140,19 @@
 
         <xi:include href="version-info.xml" xpointer="v215"/></listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>ForwardToContainers=</varname></term>
```

Comment:
i don't think this should be a global setting. 

Instead I think this should really be a per-unit setting. i.e. a container manager should enable this on the .scope unit it registers.

The thing is that different flavours of containers exist: containers-as-in-full-machines (which run systemd inside of it, maybe ssh, and thus might want to get their own crashes, too). And containers-as-in-single-services (which is this docker mess that runs a service binary as PID 1 in the container). It's up to the specific container or container manager to decide what to do with the coredump hence.

Hence, I think this should be a boolean property for all unit types that are backed by a cgroup, and should be implemented by setting some new xattr on the cgroup in cgroupfs. i.e. if "user.local-coredump" or so is set to "1" we'll try to pass the coredump to the container. see how the oomd or logging xattrs are currently maintained by pid 1.

This would then allow systemd-coredump to just search for the xattr up the cgroup tree, until it finds a cgroup that has the xattr set. It then should deliver the crash directly to the coredump socket of that container.

Reply:
Okay, that makes sense to me. I will start working on this approach. Thanks for reviewing.

Reply:
as mentioned elsewhere, let's find the outermost cgroup in the cgroup path that has user.delegate set, and then deliver the coredump to that. That's how we mark delegated cgroups.

Reply:
see is_delegated() in src/shared/cgroup-show.c for details on how to do this check properly. In fact, that function should probably be generalized and be placed in cgroup-util, so that you can just reuse it.

Reply:
The latest push does not do this exactly. Since I think we still want to join the namespaces of the container leader, it seemed to make sense to (a) find the namespace leader (like before), and then (b) check if the parent belongs to a cgroup with `Delegate=yes` and `CoredumpReceive=yes` (`CoredumpReceive=` is for the new xattr).